src/App.tsx

```diff
@@ -1,26 +1,144 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { addTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Errors } from './components/Errors/Errors';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { ErrorMessage } from './types/ErrorMessage';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredStatus, setFilteredStatus] = useState<Status>(Status.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [tempoTodo, setTempoTodo] = useState<Todo | null>(null);
+  const [title, setTitle] = useState('');
+  const [deletedTodo, setDeletedTodo] = useState(NaN);
+  const [isDeleteCompleted, setIsDeleteCompleted] = useState(false);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessage.UnableToLoad);
+        setTimeout(() => setErrorMessage(''), 3000);
+      });
+  }, []);
+
+  const handleAdd = useCallback((newTodo: Todo) => {
+    setTempoTodo(newTodo);
+    let todosLength = 0;
+
+    addTodo(newTodo)
+      .then(todo => {
+        setTodos(currentTodos => [...currentTodos, todo]);
+        todosLength = 1;
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessage.UnableToAdd);
+      })
+      .finally(() => {
+        if (todosLength === 1) {
+          setTitle('');
+        }
+
+        setTempoTodo(null);
+      });
+  }, []);
+
+  const handleDeleteCompleted = () => {
+    setIsDeleteCompleted(true);
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    Promise.allSettled(
+      completedTodos.map(todo => deleteTodo(todo.id).then(() => todo)),
+    )
+      .then(values => {
+        values.forEach(val => {
+          if (val.status === 'rejected') {
+            setErrorMessage(ErrorMessage.UnableToDelete);
+          } else {
+            setTodos(currentTodos => {
+              const todoId = val.value as Todo;
+
+              return currentTodos.filter(todo => todo.id !== todoId.id);
+            });
+          }
+        });
+      })
+      .finally(() => setIsDeleteCompleted(false));
+  };
+
+  const handleDelete = (todoId: number) => {
+    setDeletedTodo(todoId);
+
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessage.UnableToDelete);
+      })
+      .finally(() => setDeletedTodo(NaN));
+  };
```

Comment:
Let's extract reusable parts of the logic into separate functions to improve readability and  try to use async/await syntax for better readability and control flow